web/src/views/DandisetLandingView/DandisetPublish.vue

```diff
@@ -457,7 +458,8 @@ export default defineComponent({
       }
     }
 
-    async function publish() {
+    // Throttle this function so spam-clicking the button doesn't send multiple requests
+    const publish = throttle(async () => {
```

Comment:
Why not just disable the button instead, then poll to find out when the publish is finish to re-enable?

I guess the juicier question is what to expect if a bunch of people happen to all click publish on the same Dandiset at the same time. It seems like one of them would "win" the race, and the rest would receive an error message that needs to be handled?

Reply:
> Why not just disable the button instead, then poll to find out when the publish is finish to re-enable?

Throttling seemed simpler, but I think you're right that it's not enough. In addition to the race scenario you described, I hadn't considered that even in the case of a single person clicking publish, if the dandiset has a lot of assets it may take more than a second.

> It seems like one of them would "win" the race, and the rest would receive an error message that needs to be handled?

I agree, this seems like the best solution here - I'll implement this